test/mobile_app_backend/route_branching_test.exs

```diff
@@ -0,0 +1,907 @@
+        build(:route_pattern,
+          route_id: route.id,
+          typicality: :atypical,
+          representative_trip_id: trip_wickford_junction_h.id
+        )
+
+      pattern_wickford_junction_skip_canton_s =
+        build(:route_pattern,
+          route_id: route.id,
+          typicality: :atypical,
+          representative_trip_id: trip_wickford_junction_skip_canton_s.id
+        )
+
+      pattern_wickford_junction_skip_canton =
+        build(:route_pattern,
+          route_id: route.id,
+          typicality: :atypical,
+          representative_trip_id: trip_wickford_junction_skip_canton.id
+        )
+
+      pattern_wickford_junction_canonical =
+        build(:route_pattern,
+          route_id: route.id,
+          typicality: :canonical_only,
+          representative_trip_id: trip_wickford_junction_canonical.id
+        )
+
+      stop_ids = [
+        south.id,
+        back_bay.id,
+        ruggles.id,
+        hyde_park.id,
+        readville.id,
+        route128.id,
+        canton_junction.id,
+        canton_center.id,
+        stoughton.id,
+        sharon.id,
+        mansfield.id,
+        attleboro.id,
+        south_attleboro.id,
+        pawtucket.id,
+        providence.id,
+        tf_green_airport.id,
+        wickford_junction.id
+      ]
+
+      global_data =
+        Object.to_full_map([
+          south,
+          back_bay,
+          ruggles,
+          forest_hills,
+          hyde_park,
+          readville,
+          route128,
+          canton_junction,
+          canton_center,
+          stoughton,
+          sharon,
+          mansfield,
+          attleboro,
+          south_attleboro,
+          pawtucket,
+          providence,
+          tf_green_airport,
+          wickford_junction,
+          route,
+          trip_providence_hr,
+          trip_providence,
+          trip_providence_s,
+          trip_providence_r,
+          trip_providence_fhr,
+          trip_providence_fh,
+          trip_stoughton_typical,
+          trip_stoughton_h,
+          trip_stoughton_hr,
+          trip_stoughton_canonical,
+          trip_wickford_junction_typical,
+          trip_wickford_junction_s,
+          trip_wickford_junction_h,
+          trip_wickford_junction_skip_canton_s,
+          trip_wickford_junction_skip_canton,
+          trip_wickford_junction_canonical,
+          pattern_providence_hr,
+          pattern_providence,
+          pattern_providence_s,
+          pattern_providence_r,
+          pattern_providence_fhr,
+          pattern_providence_fh,
+          pattern_stoughton_typical,
+          pattern_stoughton_h,
+          pattern_stoughton_hr,
+          pattern_stoughton_canonical,
+          pattern_wickford_junction_typical,
+          pattern_wickford_junction_s,
+          pattern_wickford_junction_h,
+          pattern_wickford_junction_skip_canton_s,
+          pattern_wickford_junction_skip_canton,
+          pattern_wickford_junction_canonical
```

Comment:
There are a lot of patterns on the Providence/Stoughton Line, which makes it a good test case, but which also makes it a bad test case. I could delete more of these (I already deleted a few that are redundant here because they just go to different child stops), but then it’s not as robust of a test.